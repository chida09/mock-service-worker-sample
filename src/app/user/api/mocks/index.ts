export * from './handlers';
export * from './server';
export * from './user';

