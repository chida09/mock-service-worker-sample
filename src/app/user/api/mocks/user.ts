export const mockData = [
  {
    id: '1',
    name: 'Leanne',
    email: ''
  },
  {
    id: '2',
    name: 'Clementina',
    email: ''
  }
]
