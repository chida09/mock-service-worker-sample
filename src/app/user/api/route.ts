export async function GET() {
  const response = await fetch('https://jsonplaceholder.typicode.com/users');
  const data = await response.json();

  return Response.json(data);
}
