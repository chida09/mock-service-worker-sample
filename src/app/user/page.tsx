import { UserContainer } from './containers';

export default async function Page() {
  return (
    <>
      <UserContainer/>
    </>
  )
};
