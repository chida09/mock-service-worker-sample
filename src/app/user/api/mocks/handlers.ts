import { http, HttpResponse } from 'msw';
import { mockData } from './user';

export const handlers = [
  http.get(
    '/user/api', () => {
      return HttpResponse.json(mockData)
    }
  ),
]
