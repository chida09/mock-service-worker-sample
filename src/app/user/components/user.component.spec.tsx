import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { mockData } from '../api/mocks';
import { UserComponent } from "./user.component";

describe('UserComponent', () => {
  it('should render correctly' , () => {
    const { asFragment } = render(<UserComponent users={mockData} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render header title' , () => {
    render(<UserComponent users={mockData} />)
    expect(screen.getAllByRole('heading', { level: 1, name: 'User' })[0]).toBeInTheDocument()
  })
})
