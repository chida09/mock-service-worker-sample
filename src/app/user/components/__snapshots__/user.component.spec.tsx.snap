// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserComponent > should render correctly 1`] = `
<DocumentFragment>
  <h1>
    User
  </h1>
  <ul>
    <li>
      1: Leanne
    </li>
    <li>
      2: Clementina
    </li>
  </ul>
</DocumentFragment>
`;
